src/ucp/core/ucp_ep.c

```diff
@@ -1143,6 +1143,36 @@ static void ucp_ep_set_lanes_failed(ucp_ep_h ep, uct_ep_h *uct_eps)
     }
 }
 
+static int ucp_can_uct_ep_unprogress(ucp_ep_h ep, uct_ep_h uct_ep)
```

Comment:
better to pass `lane` here instead of `uct_ep` and rename to `ucp_ep_can_unprogress_lane`